dokuwiki-text/src/main/java/org/xwiki/contrib/dokuwiki/text/internal/input/DokuWikiInputFilterStream.java

```diff
@@ -314,6 +313,25 @@ private void readDocument(File directory, String dokuwikiDataDirectory, DokuWiki
         }
     }
 
+    private FilterEventParameters readDocument( File file ,FilterEventParameters documentLocaleParameters,
+                                                String dokuwikiDataDirectory, DokuWikiFilter proxyFilter)
+    {
+        try {
+            String pageContents = FileUtils.readFileToString(file, StandardCharsets.UTF_8);
+            String convertedContent = parseContent(pageContents);
+            documentLocaleParameters.put(WikiDocumentFilter.PARAMETER_CONTENT, convertedContent);
+            //Wiki Document Locale
+            proxyFilter.beginWikiDocumentLocale(Locale.ROOT, documentLocaleParameters);
+            readAttachments(pageContents, file, dokuwikiDataDirectory, proxyFilter);
+            return documentLocaleParameters;
+        }  catch (IOException e) {
+            this.logger.error("Could not read file", e);
+        } catch (FilterException e) {
+            this.logger.error("",e);
```

Comment:
Leaving the error may not be the best thing to do. what do you think @tmortagne we should log here?

Reply:
FilterException exceptions are not supposed catched (except in very special cases). If the output filter decided to generate a FilterException the intent is generally to stop the process so the input filter should not catch and ignore it.

Reply:
Thanks for the review @tmortagne  @slayerjain . However, I'm not sure if you meant the exception shall be left unhandled here?

Reply:
That's what I meant by not catching the exception yes. Let the exception pass trough and stop the process (of course make sure you clean everything that needs to be cleaned in a finally block but I think it's already the case).